### PR TITLE
Update API behavior for bundled assemblies

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/docs#20785.

(Trims trailing whitespace to ignore whitespace changes in diff.)

### CODE REVIEW

File: xml/System.Reflection/Assembly.xml [hunk 1]
```diff
@@ -233,22 +233,22 @@
...
- The following example uses the <xref:System.Reflection.Assembly.CodeBase%2A> property.  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+ To get the absolute path to the loaded manifest-containing file, use the <xref:System.Reflection.Assembly.Location%2A?displayProperty=nameWithType> property instead.
+
+ If the assembly was loaded as a byte array, using an overload of the <xref:System.Reflection.Assembly.Load%2A> method that takes an array of bytes, this property returns the location of the caller of the method, not the location of the loaded assembly.
+
+In .NET 5.0 and later versions, for bundled assemblies, this property throws a <System.NotImplementedException>.
```
Comment (gewarren) [inline xml/System.Reflection/Assembly.xml:243]:
```suggestion
In .NET 5.0 and later versions, for bundled assemblies, this property throws an exception.
```

File: xml/System.Reflection/Assembly.xml [hunk 2]
```diff
@@ -1578,13 +1578,15 @@
...
- The `name` should not include the path to the file.  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+ This method works on both public and private resource files.
+
+ The `name` argument should not include the path to the file.
+
+In .NET 5.0 and later versions, for bundled assemblies, this method throws a <xref:System.NotImplementedException>.
```
Comment (gewarren) [inline xml/System.Reflection/Assembly.xml:1588]:
```suggestion
In .NET 5.0 and later versions, for bundled assemblies, this method throws an exception.
```

File: xml/System.Reflection/Assembly.xml [hunk 3]
```diff
@@ -5235,20 +5237,21 @@ The ability to execute code in remote assemblies is disabled. See <see href="htt
...
-   
-  
-## Examples  
- The following example displays the location of the loaded file that contains the manifest.  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+To get the location before the file has been shadow-copied, use the <xref:System.Reflection.Assembly.CodeBase%2A> property.
```
Comment (agocke) [inline xml/System.Reflection/Assembly.xml:5244]: Is it worth noting that `CodeBase` is obsolete now? I don't think shadow copying is supported in .NET Core, so this line probably only applies to the desktop FX.

**Review by agocke — APPROVED** [review]
LGTM, except for question about desktop FX

**Pull request merged**